src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/PrivacyNoticesServiceImpl.scala

```diff
@@ -0,0 +1,56 @@
+package it.pagopa.interop.backendforfrontend.service.impl
+
+import cats.syntax.all._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.utils.TypeConversions._
+import org.scanamo.DynamoReadError.describe
+import org.scanamo._
+import org.scanamo.syntax._
+import it.pagopa.interop.backendforfrontend.service.PrivacyNoticesService
+import it.pagopa.interop.backendforfrontend.error.BFFErrors.DynamoReadingError
+import it.pagopa.interop.backendforfrontend.service.model.{PrivacyNotice, UserPrivacyNotice}
+
+import java.util.UUID
+import scala.concurrent.{Future, ExecutionContext}
+
+class PrivacyNoticesServiceImpl(tableName: String)(implicit ec: ExecutionContext, scanamo: ScanamoAsync)
+    extends PrivacyNoticesService {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  val pnTable: Table[PrivacyNotice] =
+    Table[PrivacyNotice](tableName)
+
+  val userTable: Table[UserPrivacyNotice] =
+    Table[UserPrivacyNotice](tableName)
+
+  override def getLatestVersion(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Option[PrivacyNotice]] = {
+    logger.info(s"Getting id $id privacy notice")
+    scanamo
+      .exec { pnTable.get("pk" === s"${PrivacyNotice.pkPrefix}$id" and "sk" === s"${PrivacyNotice.skPrefix}$id") }
+      .flatMap {
+        case Some(value) => value.leftMap(err => DynamoReadingError(describe(err))).toFuture.some.sequence
+        case None        => Future.successful(None)
+      }
+  }
+
+  override def getByUserId(id: UUID, userId: UUID)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Option[UserPrivacyNotice]] = {
+    logger.info(s"Getting privacy notice with id ${id.toString} for user ${userId.toString}")
+    scanamo
+      .exec {
+        userTable.get("pk" === s"${UserPrivacyNotice.pkPrefix}$id" and "sk" === s"${UserPrivacyNotice.skPrefix}$userId")
```

Comment:
same for naming and prefix
```suggestion
        userTable.get("userId" === userId and "versionNumber/versionId" === ???)
```

The partition key for this table is the `userId`.
I don't remember if the sort key was the version number or version id. @beetlecrunch ?

Reply:
See Above

Reply:
I don 't persist the history for PN versions because it is a nonsense. Only the latest version has to be persisted in my opinion. 
With the data regarding the userId it will be cleaner.

Reply:
history persistence must be agreed with @ruggerocastagnola because we could have legal requirements

Reply:
Tomorrow we have a final meeting to discuss on this matter with him

Reply:
Ruggero ask for keeping a history of acceptation for each user